bftengine/src/bftengine/Crypto.cpp

```diff
@@ -177,50 +177,43 @@ class RSAVerifierInternal {
 
 RSASigner::RSASigner(const char* privateKey) {
   StringSource s(privateKey, true, new HexDecoder);
-  d = new RSASignerInternal(s);
+  impl = std::make_unique<Impl>(s);
 }
 
-RSASigner::~RSASigner() {
-  RSASignerInternal* p = (RSASignerInternal*)d;
-  delete p;
-}
+RSASigner::RSASigner(RSASigner&&) = default;
```

Comment:
Do we want to it to be movable?

Reply:
I would say yes. For example, we can use it in standard vectors and return it from functions (without expecting RVO).